modules/cluster/irsa.tf

```diff
@@ -92,10 +92,10 @@ module "iam_assumable_role_external_dns" {
   source                        = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
   version                       = "~> v2.13.0"
   create_role                   = true
-  role_name                     = substr("tf-${var.cluster_name}-sa-role-external_dns-${local.generated_seed}", 0, 60)
+  role_name                     = var.is_jx2 ? substr("tf-${var.cluster_name}-sa-role-external_dns-${local.generated_seed}", 0, 60) : substr("${var.cluster_name}-${local.jenkins-x-namespace}-external-dns", 0, 60)
```

Comment:
Dont need sub-string.

Reply:
Switched truncation to happen per-cluster and per-namespace